python/measurement/average_taxons.py

```diff
@@ -0,0 +1,16 @@
+import gzip
+from lib import json_arrays
+from lib.helpers import dig
+from lib import services
+from statistics import mean
+
+
+def measure_average_taxons(filename):
+
+    def __number_of_taxons(content_item):
+        return len(dig(content_item, 'links', 'taxons') or [])
+
+    with gzip.open(filename, mode='rt') as file:
+        content_generator = json_arrays.read_json(file)
+        value = mean(map(__number_of_taxons, content_generator))
```

Comment:
How about doing this with lambda (I think that's a thing in Python...):

```        value = mean(map(lambda content_item: dig(content_item, 'links', 'taxons') or [], content_generator))```

or maybe laying it out like this if the line above is too long.

```
        value = mean(
            map(
                lambda content_item: dig(content_item, 'links', 'taxons') or [],
                content_generator
            )
        )
```

Reply:
I'm not sure if that makes it more readable. you forgot an additional 'len' so it would be:

`value = mean(
            map(
                lambda content_item: len(dig(content_item, 'links', 'taxons') or []),
                content_generator
            )
        )`

I think __number_of_taxons better describes what's going on

Reply:
Yeah, I take your point, giving it a name is good.